### DISCUSSION TITLE
Newbie Advice Request

### DISCUSSION THREAD

**Discussion by GaryBartlett** [joomengine/Joomla-Component-Builder#877] [General]
Hi Folks,

I'm starting out on the JCB Journey and am keen to get advice on the best way(s?) to get going.

I'm not a developer by profession, but:
1. Have developed apps in VBA, C# and PHP.
2. Can find my way around mysql, html, css & js.
3. Have been a heavy Joomla admin user for years, but haven't build a single component, yet (so know that my biggest hurdle will be learning how Joomla does what it does)!

My first component is a feature-rich cascading listview (eg. a multi-column listview of tasks, with the ability to edit the task details - each of which has a cascading listview of sub-tasks (child records), in a many-to-many relationship).

I'll build a basic JCB User Guide for Beginners app - and populate it - at the same time.

I bet that, once I've built those two, I'll be well on my way!

I've started on the JCB tutorials, but am apprehensive, given the dates on some of them, that trying to get the current version to do what was straight-forward (or accomplished differently) in the earlier version could take me down a rabbit-hole.

Where would you start, in my shoes - and how would you go about it?

My provisional approach is to watch, understand and replicate the tutorials, step-by-step, in the current version.

Thanks very much!

Gary

**Comment by ohrionmartin** [comment]
Hello there @GaryBartlett!

Welcome to the JCB community..

Although the YouTube tutorials may seem outdated, the workflow of building extensions with JCB has not changed that much. JCB offers an intuitive (to a certain level) and easy to work with interface that even though the tutorials seem old and done in an older version of JCB, you will be able to replicate that in current version of JCB by following the tutorials. 

I would encourage you to start with this [playlist](https://www.youtube.com/watch?v=1KBBtQUxMTc&list=PLQRGFI8XZ_wtGvPQZWBfDzzlERLQgpMRE) and once you've covered most of the topics you will be able develop the kind of extensions mentioned above. 

And finally, when you get stuck or struggle with something, you can always ask a question here..

**Reply by GaryBartlett** [reply]
Legend, Ohrion - thanks!

I'll do that, then.

(Starts stopwatch).
